desktop/core/src/desktop/js/apps/jobBrowser/components/impalaQueries/query-table/SortSelector.vue

```diff
@@ -0,0 +1,202 @@
+<!--
+  Licensed to Cloudera, Inc. under one
+  or more contributor license agreements.  See the NOTICE file
+  distributed with this work for additional information
+  regarding copyright ownership.  Cloudera, Inc. licenses this file
+  to you under the Apache License, Version 2.0 (the
+  "License"); you may not use this file except in compliance
+  with the License.  You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+  Unless required by applicable law or agreed to in writing, software
+  distributed under the License is distributed on an "AS IS" BASIS,
+  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  See the License for the specific language governing permissions and
+  limitations under the License.
+-->
+
+<template>
+  <dropdown-panel :text="label" :close-on-click="false">
+    <template #default="slotProps">
+      <div class="sort-selector">
+        <div class="sort-selection">
+          <div v-for="column in sortableColumns" :key="column.key" class="sort-list-entry">
+            <label>
+              <input
+                name="sort-column"
+                type="radio"
+                :checked="column.key === currentSort.column"
+                @change="setColumn(column)"
+              />
+              {{ column.label }}
+            </label>
+          </div>
+        </div>
+
+        <label class="order-selector">
+          <input
+            type="checkbox"
+            :checked="currentSort.order === 'ASC'"
+            @change="setOrder($event.target.checked)"
+          />
+          Ascending
+        </label>
+
+        <div class="sort-selector-actions">
+          <hue-link @click="cancel(slotProps.closePanel)">{{ I18n('Cancel') }}</hue-link>
+          <hue-button
+            :small="true"
+            :primary="true"
+            :disabled="applyDisabled"
+            @click="apply(slotProps.closePanel)"
+          >
+            {{ I18n('Apply') }}
+          </hue-button>
+        </div>
+      </div>
+    </template>
+  </dropdown-panel>
+</template>
+
+<script lang="ts">
+  import { defineComponent, PropType } from 'vue';
+
+  import I18n from 'utils/i18n';
+  import DropdownPanel from 'components/dropdown/DropdownPanel.vue';
+  import HueButton from 'components/HueButton.vue';
+  import HueLink from 'components/HueLink.vue';
+  import { SortInfo } from '../ImpalaQueries.vue';
+  import { Column } from 'components/HueTable';
+  import { ImpalaQuery } from '../index';
+
+  const SORTABLE_COLUMNS = new Set([
```

Comment:
Nit: Perhaps type this as Set<key of X>?

Reply:
Sorry that I don't understand. Its already typed as a set right?

Reply:
Yeah, now it's a set of any string which would allow you to add "foo" as a value even though it wouldn't work as there's no foo column. I meant that you could limit it to the strings that matches the attribute names of the structure, the available column names. But not important. :-)